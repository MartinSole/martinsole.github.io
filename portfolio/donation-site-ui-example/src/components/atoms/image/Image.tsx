interface Props {
    alt?: string
    src: string
}

const Image = ({alt = '', src}: Props) => {
  return (
    <img src={src} alt={alt} />
  )
}

export default Image