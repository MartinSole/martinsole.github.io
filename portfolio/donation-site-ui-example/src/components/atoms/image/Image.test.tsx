import '@testing-library/jest-dom'
import { getByRole, render } from "@testing-library/react"
import Image from "./Image"

test("Renders the Image component", () => {
   const {container, getByAltText, getByRole} = render(<Image src="https://placehold.co/600x400" alt="grey square"/>)
    expect(container).toMatchSnapshot()
    expect(getByRole('img')).toBeTruthy
    expect(getByAltText('grey square')).toBeTruthy
})
