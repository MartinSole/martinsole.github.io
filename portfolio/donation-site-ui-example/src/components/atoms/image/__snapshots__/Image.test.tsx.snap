// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the Image component 1`] = `
<div>
  <img
    alt="grey square"
    src="https://placehold.co/600x400"
  />
</div>
`;
